cli/azd/pkg/project/project_config_test.go

```diff
@@ -126,7 +127,38 @@ services:
 
 	require.Equal(t, "./Dockerfile.dev", service.Docker.Path)
 	require.Equal(t, "../", service.Docker.Context)
-	require.Equal(t, []string{"foo", "bar"}, service.Docker.BuildArgs)
+	require.Equal(t, []osutil.ExpandableString{
+		osutil.NewExpandableString("foo"),
+		osutil.NewExpandableString("bar"),
+	}, service.Docker.BuildArgs)
+}
+
+func TestProjectWithExpandableDockerArgs(t *testing.T) {
+	env := environment.NewWithValues("test", map[string]string{
+		"REGISTRY": "myregistry",
+		"IMAGE":    "myimage",
+		"TAG":      "mytag",
+		"KEY1":     "val1",
+		"KEY2":     "val2",
+	})
+
+	serviceConfig := &ServiceConfig{
+		Docker: DockerProjectOptions{
+			Registry: osutil.NewExpandableString("${REGISTRY}"),
+			Image:    osutil.NewExpandableString("${IMAGE}"),
+			Tag:      osutil.NewExpandableString("${TAG}"),
+			BuildArgs: []osutil.ExpandableString{
+				osutil.NewExpandableString("key1=${KEY1}"),
+				osutil.NewExpandableString("key2=${KEY2}"),
```

Comment:
nits
- What happen if the `${KEY}` is equal to something like `ff=23`?  should azd return an invalid arg error?  or would we just use the value as is.
- What happen if the separator `=` is missing?

Reply:
if the key returns a value like looks like `key=value`, then this would be interpolated before calling `docker build` and should be ok.  

If the separator is missing then it would fall back to the default behavior of `docker build` command and lookup env var matching that name.